src/flamegraph/mod.rs

```diff
@@ -250,17 +252,16 @@ impl Rectangle {
 ///
 /// [differential flame graph]: http://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html
 #[allow(clippy::cyclomatic_complexity)]
-pub fn from_sorted_lines<'a, I, W>(opt: Options, lines: I, writer: W) -> quick_xml::Result<()>
+pub fn from_sorted_lines<'a, I, W>(
+    opt: Options,
+    lines: I,
+    writer: W,
+    mut palette_map: Option<&mut color::PaletteMap<'a>>,
```

Comment:
Ouch, yeah, that's pretty unfortunate... What do we think the downside is of just making the key of `PaletteMap` `String` is?

Reply:
Well, we already allocate `String` when reading function names from the palette file. So only new functions names that are not already in the `PaletteMap` after reading the file are just borrowed. If the main use case is to use `PaletteMap` when calling `flamegraph` repetitively on inputs from different executions of the same program, we would lose virtually nothing (apart from the first iteration).

Reply:
Okay, I think I'm onboard with switching to `String` then :)